plugin/trino-hudi/src/test/java/io/trino/plugin/hudi/AbstractHudiTestQueryFramework.java

```diff
@@ -0,0 +1,180 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.trino.plugin.hudi;
+
+import com.google.common.collect.ImmutableMap;
+import io.trino.Session;
+import io.trino.plugin.hive.HivePlugin;
+import io.trino.plugin.tpch.TpchPlugin;
+import io.trino.spi.type.TimeZoneKey;
+import io.trino.testing.AbstractTestQueryFramework;
+import io.trino.testing.DistributedQueryRunner;
+import io.trino.testing.QueryRunner;
+
+import java.nio.file.Path;
+import java.util.Locale;
+import java.util.Map;
+
+import static io.trino.testing.TestingSession.testSessionBuilder;
+import static java.lang.String.format;
+
+public class AbstractHudiTestQueryFramework
+        extends AbstractTestQueryFramework
+{
+    public static final String HUDI_CATALOG = "hudi";
+    public static final String HIVE_CATALOG = "hive";
+    public static final String HUDI_SCHEMA = "default";
+
+    static final String NON_PARTITIONED_TABLE_NAME = "hudi_non_part_cow";
+    static final String PARTITIONED_COW_TABLE_NAME = "stock_ticks_cow";
+    static final String PARTITIONED_MOR_TABLE_NAME = "stock_ticks_mor";
+
+    private static final String CREATE_NON_PARTITIONED_TABLE_STATEMENT = "CREATE TABLE %s.\"%s\".\"%s\" (\n" +
+            "    _hoodie_commit_time varchar,\n" +
+            "    _hoodie_commit_seqno varchar,\n" +
+            "    _hoodie_record_key varchar,\n" +
+            "    _hoodie_partition_path varchar,\n" +
+            "    _hoodie_file_name varchar,\n" +
+            "    rowid varchar,\n" +
+            "    partitionid varchar,\n" +
+            "    precomb bigint,\n" +
+            "    name varchar,\n" +
+            "    versionid varchar,\n" +
+            "    tobedeletedstr varchar,\n" +
+            "    inttolong integer,\n" +
+            "    longtoint bigint\n" +
+            " )\n" +
+            " WITH (\n" +
+            "    external_location = '%s',\n" +
+            "    format = 'PARQUET'\n" +
+            " )";
+
+    private static final String CREATE_PARTITIONED_TABLE_STATEMENT = "CREATE TABLE %s.\"%s\".\"%s\" (\n" +
+            "    _hoodie_commit_time varchar,\n" +
+            "    _hoodie_commit_seqno varchar,\n" +
+            "    _hoodie_record_key varchar,\n" +
+            "    _hoodie_partition_path varchar,\n" +
+            "    _hoodie_file_name varchar,\n" +
```

Comment:
what are those?
Internal hudi columns should not be part of `CREATE TABLE` statements. They should be automatically added to table schemas like we do in Hive or Delta. 
See: https://github.com/trinodb/trino/blob/220c0f85df6a7b60a76756813c2d5f6dc1ba77c5/plugin/trino-hive/src/main/java/io/trino/plugin/hive/util/HiveUtil.java#L862-L873